src/users/users.controller.ts

```diff
@@ -2,37 +2,79 @@ import { Controller, Get, Post, Body, Param, Patch, Delete } from '@nestjs/commo
 import { UsersService } from './users.service';
 import { CreateUserDto } from './dto/create-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
+import { ApiExtraModels, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { User } from './entities/user.entity';
-import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
 
 @ApiTags('User Details')
 @Controller('users')
+@ApiExtraModels(User)
 export class UsersController {
   constructor(private readonly usersService: UsersService) {}
 
-  @Post()
-  @ApiResponse({ status: 201, description: 'The record has been successfully created.' })
-  @ApiResponse({ status: 400, description: 'Bad Request.' })
-  async create(@Body() createUserDto: CreateUserDto): Promise<User> {
-    return this.usersService.create(createUserDto);
+  @Get(':username')
+  @ApiResponse({
+    status: 200,
+    description: 'A User objects.',
+    schema: { $ref: '#/components/schemas/User' },
+  })
+  @ApiResponse({ status: 401, description: 'Authorization information is missing or invalid.' })
+  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  @ApiResponse({ status: 404, description: 'A user with the specified username was not found.' })
+  @ApiResponse({ status: 500, description: 'Unexpected error.' })
+  async findOne(@Param('username') username: string): Promise<User> {
+    return await this.usersService.findByUsername(username);
   }
 
   @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'List of User objects.',
+    schema: {
+      type: 'array',
+      items: {
+        $ref: '#/components/schemas/User',
+      },
+    },
+  })
+  @ApiResponse({ status: 401, description: 'Authorization information is missing or invalid.' })
+  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  @ApiResponse({ status: 404, description: 'There is no user.' })
+  @ApiResponse({ status: 500, description: 'Unexpected error.' })
   async findAll(): Promise<User[]> {
     return await this.usersService.findAll();
   }
 
-  @Get(':username')
-  async findOne(@Param('username') username: string): Promise<User> {
-    return await this.usersService.findByUsername(username);
+  @Post()
+  @ApiResponse({ status: 200, description: 'The user has been successfully updated.' })
+  @ApiResponse({ status: 201, description: 'The user has been successfully created.' })
```

Comment:
Ici le user est retourné vu la fonction, on peut faire comme tu as fait pour l'afficher ducoup ? 
```suggestion
  @ApiResponse({ status: 201, description: 'The user has been successfully created.',
    schema: { $ref: '#/components/schemas/User' },
  })
```

Reply:
(Pareil sur book)